ExtTinker App (RGB-LED control, RX/TX digitalRead/Write)

ScruffR opened the pull request:
This version of Tinker App extends the original allowing to control the on-board RGB-LED and the RX/TX pins when they are not used as USART pins.
To work as intended the Spark Core firmware has to be updated to this version https://dl.dropboxusercontent.com/u/60050879/TinkerFW.txt

See preview at https://community.spark.io/t/extended-android-spark-tinker-app-rgb-led-rx-tx-pins/2899

ScruffR committed:
Added new drawables for RGB

ScruffR committed:
Added RX/TX digiatlRead/Write support
Added RGB-LED control capability (CL,<argb colorcode>)
Fixed some warings
Requires now VIBRATE privilege to give feedback for color control

towynlin commented:
Thanks @ScruffR! After discussing this a bit at Spark, we think the redesign here doesn't fit our design standards well, plus it brings the iOS app and android app out of alignment. Also, it makes the app harder to understand for beginners, and being super easy for those with no previous hardware experience (e.g., have never heard of "serial") is one of the main goals of Tinker.

For those reasons, we won't pull this into the Spark repo. However, we definitely encourage you to advertise your fork of the android app and tinker firmware!

Definitely keep pull requests that are useful to everybody coming! We really appreciate your work! :smile_cat:

ScruffR commented:
After some mail discussion with Zachary, I think I was able to put most of above "reasons" into perspective.

So I hope this PR can get reviewed once more - taking my arguments into consideration.

The only undisputed reason is that the iOS version of the app would fall behind the Android version. But this is mainly due to unavailability of the iOS sources :wink:

towynlin commented:
I still haven't found the time to do a full review here, but yes, ScruffR convinced me that it's worth a closer look.

ScruffR commented:
Thanks Zachary, happy to hear this :smile:

Please let me know if anything needs further adaption to meet your requirements. 
After all, it will still be your product, so you have the final word how things should look and how it has to act.
I have already found some minor issues (e.g. log complains about wrong return code, when analogWrite() function passes back the ARGB value) and I am also experimenting with touch rotating the color wheel (like an old fashioned phone dial) rather than having a static color wheel.
Another experimental feature is to use the camera to pick the color for the RGB-LED.

So the whole thing is still work in progress, and anything you find wrong or would like to see, could be corrected/added - I guess :wink:

ScruffR committed:
Fixed pin colors for analogRead/analogWrite HIGH (same as LOW)
Color selector shadow stayed visible after onClick of RGB-LED symbol -
fixed.

ScruffR commented:
Hello Zachary ( @towynlin ) again!
I've done some cleanup (resolution independence of instruction screen, optical balance for TX/RX pins, color wheel with shadow, stopped sliding pane from sliding while color selecting, ...) and added some features (rotating color wheel for saturation/volume control of tap-selected hue, per pin reset/hide feature on mode view).

I've not issued a PR but I have uploaded an APK on my DropBox for you to review before I do that 

https://dl.dropboxusercontent.com/u/60050879/ExtTinkerApp.apk

I'd like to hear what you are thinking of this.

Thanks

Andy

towynlin commented:
Thanks much Andy. I'm sorry this has been so low on the priority list. I look forward to reviewing it when I get the time. Thanks for the dropbox APK, that will make testing faster when I get to it.

ScruffR committed:
Preparations for alternate API host server

ScruffR committed:
Fixed pin mode selector dialog (pin name smaller font, hide/reset
smaller, hide/reset color bar wrong side)

ScruffR committed:
Preparations for alternate API host server

ScruffR committed:
Alternate API host server support
Added a new AlterHostActivity to configure API URL scheme, host server
name, host server port (accessible via SignUpActivity & LoginActivity).
Configured values are stored via Prefs and are used in AppConfig and
ApiUrlHelper.
Minor layout adaptions.

ScruffR committed:
Per Core setting if "legacy mode" or "ExtTinker features" is required.
Removed currently unneeded camera permissions from manifest.
Added hint on 'activity_alter_host', to suggest http and port 8080 for
easy testing of local cloud.

ScruffR commented:
Just added some more features (local cloud support, legacy mode to avoid confusion for non ExtTinker firmware Cores) - incorporated in above APK link.

towynlin commented:
Thanks! We're in the process of hiring a mobile developer — when that person comes on board, you can expect this to get some attention. You rock!

ScruffR commented:
Thanks! 
I hope he doesn't cry out loud, when he sees, how I did it ;-)
